javascript/core/src/engine/verbs/convert.ts

```diff
@@ -26,19 +27,46 @@ const doConvert = wrapColumnStep<ConvertArgs>(
 export const convert = makeStepFunction(doConvert)
 export const convertNode = makeStepNode(doConvert)
 
-function parseType(column: string, type: ParseType, radix?: number) {
+function parseType(
+	column: string,
+	type: ParseType,
+	radix?: number,
+	formatPattern?: string,
+) {
+	const parseTime = timeParse(formatPattern ?? '%Y-%m-%d')
+	const formatTime = timeFormat(formatPattern ?? '%Y-%m-%d')
+
 	return escape((d: any) => {
 		const value = d[column]
 		switch (type) {
 			case ParseType.Boolean:
 				// arquero has no boolean operation
 				return bool(value)
-			case ParseType.Date:
-				return op.parse_date(value)
+			case ParseType.Date: {
+				if (value !== null && !isNaN(value)) return new Date(value)
+
+				return formatPattern === '%Y-%m-%dT%H:%M:%S.%LZ'
+					? isoParse(value)
+					: parseTime(value)
+			}
 			case ParseType.Integer:
 				return op.parse_int(value, radix)
 			case ParseType.Decimal:
 				return op.parse_float(value)
+			case ParseType.String: {
+				if (
+					typeof value === 'string' &&
+					value.trim().toLowerCase() === 'undefined'
```

Comment:
I think lodash isNil could simplify this check

Reply:
Ah, I see. Ok, let's leave that as is. It might change with the new data types rules, but I'd rather go through all of those in one sweep for consistency anyway.